modules/node_modules/@ncigdc/components/File.js

```diff
@@ -54,19 +65,36 @@ const File = ({ node, theme, searchTerm, setState }) => {
     </Link>)
     : '--';
 
-  node.tags = node.tags ? JSON.parse(node.tags) : '[--]';
+  node.tags = node.tags && node.tags !== '[--]' ? JSON.parse(node.tags) : '[--]';
 
   return (
     <Column>
-      <Row style={{ justifyContent: 'space-between', padding: '1rem 0' }}>
-        <Column style={{ justifyContent: 'center', marginLeft: 'auto' }}>
-          <AddToCartButtonSingle
-            style={{ height: '30px', backgroundColor: theme.primary, color: 'white' }}
-            file={node}
-            addText="Add to Cart"
-            removeText="Remove from Cart"
-          />
-        </Column>
+      <Row style={{ padding: '1rem 0', justifyContent: 'flex-end' }}>
+        <Button
+          onClick={() => dispatch(toggleFilesInCart(node))}
+          leftIcon={<ShoppingCartIcon />}
+        >
+          {fileInCart(files, node) ? 'Remove from Cart' : 'Add to Cart'}
+        </Button>
+        {
+          userCanDownloadFile({ user, file: node }) ? (
+            <DownloadButton
+              extraParams={{ ids: node.file_id }}
+              filename={node.file_name}
+              url={`${API}data?annotations=true&related_files=true`}
```

Comment:
I think for downloading files, should use the auth api endpoint, otherwise controlled files wouldn't get work. Like this: https://github.com/NCI-GDC/portal-ui/pull/1254/files#diff-3a31735e0cbeacf6375e5379ae9742aaR114, I added the GDC_AUTH env var on my PR